src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java

```diff
@@ -2111,6 +2111,14 @@ protected String extractAuthorName(Document doc) {
                 }
             }
 
+            if (authorName.isEmpty()) { // hack for http://mspmentor.net
+                result = doc.select("span[class=author-name]").first();
+                if (result != null) {
+                    authorName = SHelper.innerTrim(result.ownText());
+                    if(DEBUG_AUTHOR_EXTRACTION && !authorName.isEmpty()) System.out.println("AUTHOR: span[class=authorname]");
```

Comment:
Good catch @skyshard that why it didn't match with the hack for https://github.com/skyshard/snacktory/blob/67804f98d90e1dc52e28c308f9d9cbc3af3ef8f0/src/main/java/de/jetwick/snacktory/ArticleTextExtractor.java#L2084

On further debugging found that jsoup select
span [class$=author-name] is not really equals span[class$=author-name]. Note the *blank space* between span and [class$=author-name].
----
About combining common rules, I have combined some of the selectors as you suggested. But there is a constraint how much we can rearrange or club the rules since sequence of rule really plays the important role here. Changing the rule position breaks several articles.